### PR TITLE
Feat/v3/bulk edit permission rights language

### PR DESCRIPTION
**Pull request by le0m**
This PR adds the ability to edit language, rights and permissions in bulk for multiple objects.

### CODE REVIEW

File: bedita-app/app_controller.php [hunk 1]
```diff
@@ -1238,6 +1238,62 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
         $this->setSessionForObjectDetail($objects['items']);
     }
 
+    public function changeLanguageObjects() {
```
Comment (didoda) [inline bedita-app/app_controller.php]: add a basic phpdoc ;)

File: bedita-app/app_controller.php [hunk 2]
```diff
@@ -1238,6 +1238,62 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+        if (!empty($this->params['form']['objects_selected'])) {
+            $objectsToModify = $this->params['form']['objects_selected'];
+            $this->changeBulkObjects($objectsToModify, 'lang', $this->data['lang']);
+        }
+    }
+
+    public function changeRightsObjects() {
```
Comment (didoda) [inline bedita-app/app_controller.php]: add a basic phpdoc ;)

File: bedita-app/app_controller.php [hunk 3]
```diff
@@ -1238,6 +1238,62 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+            }
+
+            $beObject->id = $id;
+
+            if (!$beObject->saveField($field, $value)) {
+                throw new BeditaException(sprintf(__('Error saving %s for item: %s', true), $field, $id));
+            }
+        }
+
+        $this->Transaction->commit();
```
Comment (didoda) [inline bedita-app/app_controller.php:1295]: no rollback handled.
I'd rather use a `try / catch` block, with transaction rollback, on catch

File: bedita-app/app_controller.php [hunk 4]
```diff
@@ -1238,6 +1238,62 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+    }
+
+    public function addPermissionsObjects() {
```
Comment (didoda) [inline bedita-app/app_controller.php]: add a basic phpdoc ;)

File: bedita-app/app_controller.php [hunk 5]
```diff
@@ -1238,6 +1238,62 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+        if (!empty($this->params['form']['objects_selected'])) {
+            $objectIds = $this->params['form']['objects_selected'];
+            $this->Transaction->begin();
+
+            foreach ($objectIds as $id) {
+                $permission = ClassRegistry::init('Permission');
+                $permission->add($id, $this->data['Permission']);
+            }
+
+            $this->Transaction->commit();
```
Comment (didoda) [inline bedita-app/app_controller.php:1314]: handle rollback

**Review by didoda — CHANGES_REQUESTED** [review]

File: bedita-app/app_controller.php [hunk 6]
```diff
@@ -1238,6 +1238,86 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
         $this->setSessionForObjectDetail($objects['items']);
     }
 
+    /**
+     * Change "lang" field in bulk to objects.
+     *
+     * Used by "list_objects_bulk.tpl" view element.
+     *
+     * @throws BeditaException
```
Comment (didoda) [inline bedita-app/app_controller.php]: not necessary in here, it's already in `changeBulkObjects` phpdoc

File: bedita-app/app_controller.php [hunk 7]
```diff
@@ -1238,6 +1238,86 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+            $this->changeBulkObjects($objectsToModify, 'lang', $this->data['lang']);
+        }
+    }
+
+    /**
+     * Change "rights" field in bulk to objects.
+     *
+     * Used by "list_objects_bulk.tpl" view element.
+     *
+     * @throws BeditaException
```
Comment (didoda) [inline bedita-app/app_controller.php]: as above, not necessary in here, it's already in `changeBulkObjects` phpdoc

**Review by didoda — APPROVED** [review]

File: bedita-app/app_controller.php [hunk 8]
```diff
@@ -1238,6 +1238,86 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+    protected function changeBulkObjects($objectIds, $field, $value)
+    {
+        $beObject = ClassRegistry::init('BEObject');
+        $this->Transaction->begin();
+
+        foreach ($objectIds as $id) {
+            $this->checkObjectWritePermission($id);
+
+            if ($beObject->isFixed($id)) {
+                throw new BeditaException(sprintf(__('Error: changing %s to a fixed object!', true), $field));
```
Comment (le0m) [inline bedita-app/app_controller.php:1285]: Is this ok? Should I run gettext to generate pot files?
This text string has a variable in it, meaning that a translation is needed for all possible values of `$field`.

File: bedita-app/app_controller.php [hunk 9]
```diff
@@ -1238,6 +1238,86 @@ protected function paginatedList($id, $filter, $order, $dir, $page, $dim) {
...
+            }
+
+            $beObject->id = $id;
+
+            if (!$beObject->saveField($field, $value)) {
+                throw new BeditaException(sprintf(__('Error saving %s for item: %s', true), $field, $id));
```
Comment (le0m) [inline bedita-app/app_controller.php:1291]: Same as above

File: bedita-app/app_controller.php (hunk 8 above)
Comment (didoda) [reply bedita-app/app_controller.php:1285]: yes, it's ok

Comment (didoda) [reply bedita-app/app_controller.php:1285]: yes, you should relaunch `./cake.sh gettext update`

**Pull request merged**